src/lib.ts

```diff
@@ -1,17 +1,24 @@
 import * as core from '@actions/core'
 import * as exec from '@actions/exec'
+import { ExecOptions } from '@actions/exec'
 import * as io from '@actions/io'
 
+async function execBC(arg: string): Promise<void> {
```

Comment:
The name you chose - `execBC` has an acronym (not a great practice) and implies that it just executes buildcache - but that is not its real purpose. The whole point of the method is not to execute buildcache (buildcache was already being executed...) but to execute buildcache without impersonation

```suggestion
async function execBuildcacheWithoutImpersonation(arg: string): Promise<void> {
```

That is a much longer name, but now for any future developer the name explain what the method does. The developer no longer needs to read the method to understand it, if they see it being called somewhere